Update all translations, standardize/normalize new/open/save file menu terminology in UI and code

simonrepp opened the pull request:
Hey,

So originally I just had in mind to create a small UX related PR that renames "**Load** scene ..." in the file menu into the widely used and easier spotted "**Open** scene ..." and puts it **before** the "Save scene (...)" menu entries, as is also much more common.

Then I got a little carried away and completed all the missing spanish translations too, also correcting and improving some german and english ones too where I spotted issues. Plus I also extended the interface change for the file menu a bit into code, because I noticed that the terminology was very inconsistent there, probably a side-effect of organic growth, I know how that goes. :)

I've marked this WIP for now because it does not run for me currently, and I haven't yet methodically determined whether that is because of the changes in my PR or because I didn't check if I had a clean build and run state before I started making changes in the first place. :D Will remove the WIP label as soon as that is clarified on my side haha.

simonrepp commented:
Ok resolved everything, did some more UI research on how the vast majority of applications design the file menu and went and refined the menu some more, namely taking out the (sometimes present, sometimes not) "scene" addendum of menu entries altogether and just boiling it down to the essentials of what one is looking for in the menu. Visual comparison of the change below.

If this looks and reviews fine for you, this is generally mergeable now (as in: successfully tested to run ;)).

`Before` (left) => `After` (right)

![change_visual](https://user-images.githubusercontent.com/13210/102658026-112ce280-4177-11eb-92c5-1b70419c814f.png)

simonrepp commented on ts/omm_de.ts:
```diff
@@ -1398,7 +1398,7 @@ Soll die Selektion trotzdem entfernt werden?</translation>
     <message>
         <source>The name of the object to render. Keep clear to render the whole scene. The name does not need to be unique. In this case, all matching objects are rendered. Perl-like regular expressions are supported.</source>
-        <translation>Der Name des zu rendernden Objektes. Leer lassen um die ganze Szene zu rendern. Der Pfad muss nicht unbedingt eindeutig sein. In diesem Fall werden alle passenden Objekte gerendert. Perl-artige reguläre Ausdrücke werden unterstüzt.</translation>
+        <translation>Der Name des zu rendernden Objektes. Leer lassen um die ganze Szene zu rendern. Der Name muss nicht unbedingt eindeutig sein. In diesem Fall werden alle passenden Objekte gerendert. Perl-artige reguläre Ausdrücke werden unterstüzt.</translation>
```
Sure thing, if it's a path then it's a path. :)

I changed this because there is another, almost identical translation/source text like this where only `name` is substituted by `path`, so I figured they need to be translated differently to indictate some difference. 

For me the deeper question is maybe more the following though: Below are the two source texts related to this, coming from two different command line arguments specified in `subcommandlineparser.cpp`, but seeing these for the first time, I don't really understand how they differ, they seem to be the same (if like you say above, `name` is also a path type argument)? 

https://github.com/pasbi/ommpfritt/blob/8c90ef97a86720db6eaa698d4831ed76a579f949/src/subcommandlineparser.cpp#L41-L50

https://github.com/pasbi/ommpfritt/blob/8c90ef97a86720db6eaa698d4831ed76a579f949/src/subcommandlineparser.cpp#L57-L66

pasbi replied:
You can see the difference here:
https://github.com/pasbi/ommpfritt/blob/8c90ef97a86720db6eaa698d4831ed76a579f949/src/maincli.cpp#L108

Note that it's forbidden to set both options at the same time, at least of them must not be set.
If the `object`-option is set, only the object name is matched, if the `path`-option is set, the object path is matched.

I find this makes sense, but I agree that the documentation could be more explicit on that.
One could argue that one could match always against the object path, and use the regex `'^_root_/.*/ObjectName$'` to match arbitrary paths.
From that perspective, the `object` option is just a convenient shortcut that is currently not used and might be removed if it creates too much confusion.
The `path` option, however is required to render the icons (see [generate-icons.py](https://github.com/pasbi/ommpfritt/blob/8c90ef97a86720db6eaa698d4831ed76a579f949/build-scripts/generate-icons.py)).

simonrepp replied:
Thanks for clarifying, if that's how it's intended the documentation is actually sufficient and clear - but the name argument then really *is* a name argument, not a path (that's what confused me - it's maybe a "name pattern", a "name expression" or a "name wildcard" if so supplied, but as I understand it now with your explanation it's definitely never a path for the `--object` option - please elaborate further if I still got this wrong somehow now :)).

pasbi replied:
maybe it suffices if we

 1. change the "valueName" (the third argument, currently  `QObject::tr("NAME")` to `PATTERN` for both options
 2. make the option names more consistent: `object` -> `object-name` and `path` -> `object-path`.

simonrepp replied:
I added another commit to differentiate and (hopefully correctly :)) clarify the argument placeholders in the commandline documentation.

simonrepp replied:
Ah whoops sorry missed out on your post inbetween ... 1. I did more or less, of 2. I'm actually a bit wary because it will require further changes in more places to not break the icon generation e.g. for automated builds?

pasbi replied:
regarding 1.: I'd prefer something with PATTERN, not REGEX, I think that's more standard (and technically, the regex is compiled by the application, the user only sets its pattern).

regarding 2.: no it won't because the icon generation uses the short command names (`-p`).
But updating the icon generation wouldn't be a big deal, if necessary.
Don't worry too much, if you break something, the automatic tests should fail and tell you what's wrong :wink:

pasbi commented on src/subcommandlineparser.cpp:
```diff
@@ -45,7 +45,7 @@ static const std::map<QString, QList<QCommandLineOption>> options{
                       "In this case, all matching objects are rendered. "
                       "Perl-like regular expressions are supported."),
-          QObject::tr("NAME"),
+          QObject::tr("NAME_REGEX"),
```
better `NAME_PATTERN` ?

simonrepp replied:
I used regex over pattern because it's more precise in meaning. (Rationale: If someone does not know what REGEX is, they will likely not know what to do with PATTERN either, but someone who would understand REGEX, might be in doubt how to understand PATTERN - glob wildcard? regex? ...) But I'll happily let myself be convinced over PATTERN if I missed something in my rationale. :)

pasbi replied:
Looking at the [man page of `locate`](https://linux.die.net/man/1/locate), you see both `REGEXP` and `PATTERN` being used as the variable name.
`PATTERN` only if both globbing and regex are possible and `REGEXP` if there's no option for globbing.
That's seconding your rationale, so I'm fine with that.

simonrepp replied:
Cool appreciate the research :+1:

pasbi commented on src/subcommandlineparser.cpp:
```diff
@@ -61,7 +61,7 @@ static const std::map<QString, QList<QCommandLineOption>> options{
                       "In this case, all matching objects are rendered. "
                       "Perl-like regular expressions are supported."),
-          QObject::tr("NAME"),
+          QObject::tr("PATH_REGEX"),
```
better `PATH_PATTERN` ?

pasbi commented on src/subcommandlineparser.cpp:
```diff
@@ -61,7 +61,7 @@ static const std::map<QString, QList<QCommandLineOption>> options{
                       "The path does not need to be unique. "
                       "In this case, all matching objects are rendered. "
```
While you're at it, I think lines 61 and 62 should be one sentence, separated by a comma. The "In this case" is superfluous, IMHO. Same for lines 45, 46.

simonrepp replied:
Will do - question in between: up to now it seemed to automagically rebuild the `.ts` files, and spontaenously stopped now (likely because I'm not making changes that trigger the rebuild), I've looked at the build documentations and a bit at the cmake files etc. but I can't really figure out what's the one command I'm supposed to run to get all `.ts` files updated - do you have that in your head by any chance?

simonrepp replied:
Question resolved meanwhile, I extracted the `lupdate ...` command from somewhere and that one worked.

pasbi replied:
Yeah, the translation generation is quite complex.
The problem is a circular dependency.
If you want to ensure that your translations are re-generated, you may want to remove all `*.qm` files and all compiled resource files (`qrc_resources*.cpp{,.o}`).
That of course does not save you from using `lupdate` to edit the translations manually.

simonrepp commented:
I added the requested CLI argument changes, plus a few minor translations improvements (elsewhere) that I spotted inbetween. :+1:

pasbi commented on src/subcommandlineparser.cpp:
```diff
@@ -39,29 +39,27 @@ static const std::map<QString, QList<QCommandLineOption>> options{
-                      "The name does not need to be unique. "
-                      "In this case, all matching objects are rendered. "
+          {"q", "object-name"},
+          QObject::tr("The name of the object to render."
```
The space at the end of the strings was not a mistake.
If the strings are concatenated, the space is required, IIRC.

simonrepp replied:
Oh yes, didn't realize they're concatenated, thanks for pointing it out.

pasbi commented:
Do you have an idea why the builds fail? It seems to be some issue with numpy, but I don't see that you have modified anything that could cause the build to fail?

https://travis-ci.org/github/pasbi/ommpfritt/builds/750585369

I've restarted the builds. Sometimes Travis is just weird, and restarting the build is all it needs to fix it.

simonrepp commented:
I found these 3 lines at the very end of the report on the failing step:

```
[warning] 2020-12-19T19:31:04 :0: QCommandLineParser: option not defined: "object"
[warning] 2020-12-19T19:31:04 :0: QCommandLineParser: option not defined: "path"
[critical] 2020-12-19T19:31:04 :0: Expected exactly one matching object but found 77.
```

It seems although the automated build uses the short-form `-p` flag, the command line parser internally converts the argument key to long-form for further processing, and there was yet another place in the code, in another file, where the long-form arguments additionally needed to be changed, which I didn't realize.

In the end, as you predicted, the CI caught this. :)

I confirmed the error locally, fixed this, and confirmed the CLI call runs locally again too, also I rebased the PR on master again, so CI should pass now, fingers crossed.
